### PR TITLE
ref(saved-search): Use CompactSelect

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before**
Screen Shot 2022-05-04 at 12 49 46 PM

**After**
Screen Shot 2022-05-04 at 12 49 32 PM

### CODE REVIEW

File: static/app/views/issueList/savedSearchMenu.tsx [hunk 1]
```diff
@@ -1,238 +0,0 @@
-import {Fragment} from 'react';
```
Comment (scttcper) [inline static/app/views/issueList/savedSearchMenu.tsx:1]: yeah probably a good idea to go back to having it all in one file, this was split into two files when we were rolling out the "for review" feature

File: static/app/views/issueList/savedSearchTab.tsx [hunk 2]
```diff
@@ -35,111 +42,179 @@ function SavedSearchTab({
...
-    margin-top: ${space(1)};
-    min-width: 20vw;
-    max-width: 25vw;
-    z-index: ${p => p.theme.zIndex.globalSelectionHeader};
-
-    :after {
-      border-bottom-color: ${p => p.theme.backgroundSecondary};
-    }
+const StyledCompactSelect = styled(CompactSelect)<{isActive?: boolean}>`
+  && {
```
Comment (scttcper) [inline static/app/views/issueList/savedSearchTab.tsx:178]: what is `&&`

**Review by scttcper — APPROVED** [review]
looks good, the checkmark position feels a bit funky 
image

File: static/app/views/issueList/savedSearchTab.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/views/issueList/savedSearchTab.tsx:178]: oh it's just a double selector to override the navigation tab styles:
Screen Shot 2022-05-04 at 1 05 01 PM

File: static/app/views/issueList/savedSearchTab.tsx [hunk 3]
```diff
@@ -35,111 +42,179 @@ function SavedSearchTab({
...
+        value: 'global-searches',
+        label: t('Recommended Searches'),
+        options: globalSearches.map(search => getSearchOption(search, 'global-search')),
+      });
+    }
+
+    return searchOptions;
+  }, []);
+
+  const trigger = ({props, ref}) => (
```
Comment (evanpurkhiser) [inline static/app/views/issueList/savedSearchTab.tsx:125]: No types here?

File: static/app/views/issueList/savedSearchTab.tsx [hunk 4]
```diff
@@ -35,111 +42,179 @@ function SavedSearchTab({
...
         </Fragment>
       ) : (
         t('Saved Searches')
       )}
-    </TitleWrapper>
+    </StyledDropdownTrigger>
+  );
+
+  const onChange = useCallback(
+    option => {
```
Comment (evanpurkhiser) [inline static/app/views/issueList/savedSearchTab.tsx:150]: No types here?

**Pull request merged**